src/julia.h

```diff
@@ -87,7 +92,7 @@ DLLEXPORT void jl_threading_profile(void);
 #  define JL_ATOMIC_FETCH_AND_ADD(a,b)                                    \
        _InterlockedExchangeAdd((volatile LONG *)&(a), (b))
 #  define JL_ATOMIC_COMPARE_AND_SWAP(a,b,c)                               \
-       _InterlockedCompareExchange64(&(a), (c), (b))
+       _InterlockedCompareExchange64((volatile LONG64 *)&(a), (c), (b))
```

Comment:
i guess i would slightly rather have the cast on the comparison (casting the lhs to a uv_thread_t) rather than the pointer. the failure behavior is a bit more localized in the event the definition changes.